### ISSUE TITLE
Unable to remove handlers

### ISSUE THREAD

**Issue by JoshAshby** [piotrmurach/tty-logger#14]
### Describe the problem
Removing a handler by its symbol, such as is shown in the README, doesn't actually remove the handler.

### Steps to reproduce the problem
```ruby
#!/usr/bin/env ruby
# frozen_string_literal: true

require "tty-logger"

logger = TTY::Logger.new do |config|
  config.handlers = [
    [:console, output: $stderr]
  ]
end

logger.info "Should log"

res = logger.remove_handler :console
puts "res is nil? #{ res.nil? }"

logger.info "Shouldn't log, but does"
```

### Actual behaviour
```
ℹ info    Should log
res is nil? true
ℹ info    Shouldn't log, but does
```

Additionally, if you inspect the logger instance, `@ready_handlers` has the console handler even after the call to `#remove_handler`, which aligns with the result being `nil` and not the deleted array item.

### Expected behaviour
The second `#info` shouldn't have been logged to the console.

### Describe your environment

* OS version: MacOS 10.15.7
* Ruby version: 2.7.1
* TTY::Logger version: 0.5.0

**Comment by JoshAshby** [comment]
This seems to be caused by `#remove_handler` calling `@ready_handlers.delete` with the provided symbol, however `@ready_handlers` is an Array of instances that won't evaluate to be equal to the symbol.

https://github.com/piotrmurach/tty-logger/blob/92d603dc38586843f99da0772df3d05a9dcbf27a/lib/tty/logger.rb#L153-L155


https://github.com/piotrmurach/tty-logger/blob/92d603dc38586843f99da0772df3d05a9dcbf27a/lib/tty/logger.rb#L103

Indeed, getting the instance of the handler and passing that to `#remove_handler` does work as expected but is less than ideal since `@ready_handlers` doesn't have a public attribute accessor:

```ruby
handlers = logger.instance_variable_get :@ready_handlers
console_handler = handlers.find { _1.is_a? TTY::Logger::Handlers::Console }
logger.remove_handler(console_handler)
```

I think it'd be enough to change `#remove_handler` to use `#delete_if` and `#coerce_handler` which would match the existing behavior of `#remove_handler` if `ready_handlers` was an array of symbols:

```ruby
def remove_handler(handler)
  handler_type = coerce_handler(handler)
  @ready_handlers.delete_if { |handler| handler.is_a? handler_type }
end
```

Happy to put together a PR for this and some tests for it if this seems like the appropriate change (I only found [one test](https://github.com/piotrmurach/tty-logger/blob/94796b2b4c4e0a127ce0bd91f7292de85e2ab912/spec/unit/add_handler_spec.rb#L7-L24) using `#remove_handler` and it does not appear to actually test if the handler was removed).

**Comment by piotrmurach** [comment]
Hi Josh 👋 

Thank you for using `tty-logger` and sorry for the late reply! There is something wrong with notifications for this project. I don't get any. Anyhow, this sounds like a desirable change. Please could you submit PR with tests?

**Comment by piotrmurach** [comment]
Released `v0.6.0` that fixes this issue.
